### ISSUE TITLE
Adding elements to SyncList on server creates twice as many elements on clients

### ISSUE THREAD

**Issue by GasimoCodes** [FirstGearGames/FishNet#769]
**General**
Unity version: 6000.0.16f1
Fish-Networking version: Both stable and 4.4.5
Discord link: https://discord.com/channels/424284635074134018/1034477094731784302/1283181700419026975

**Description**
Calling SyncList.Add on from OnStartServer() adds element to list on server once and on client twice.

**Replication**
1. Create script inheriting `NetworkBehaviour`
2. Add synclist: 
```csharp
private readonly SyncList<SomeClass> items = new SyncList<SomeClass>();
```
3. Give ownership over this object to client and then add item to `SyncList`:
```csharp
        public override void OnStartServer()
        {
            base.OnStartServer();
            base.GiveOwnership(SomeNetworkedObject.Owner);
            items.Add(null);
        }
```
4. Observe calls from OnChanged events on both server and client. Client receives add operation twice, resulting in doubled SyncList.
 

**Expected behavior**
Client SyncList should have the same amount of elements as the server.

**Screenshots**
Host OnChanged Log:
![image](https://github.com/user-attachments/assets/1efbe519-b7e7-4afa-a7b5-b74a61402be1)
Client OnChanged Log:
![image](https://github.com/user-attachments/assets/3b0184d2-2ec2-4ea0-83d5-bfa32b6fa3af)

**Comment by FirstGearGames** [comment]
Have you tested this in Beta on 4.4.5?

**Comment by FirstGearGames** [comment]
I tried the following code on a scene object and all worked properly. I also tried using **OnStartServer** (without giving ownership) and it also worked.

Values were correct on server only, clientHost, and client only.

```cs
using FishNet.Connection;
using FishNet.Object;
using FishNet.Object.Synchronizing;
using UnityEngine;

public class SomeClass { }

public class SyncList_Count_Test : NetworkBehaviour
{
    private readonly SyncList<SomeClass> _st = new();

    private void Awake()
    {
        _st.OnChange += StOnOnChange;
    }

    public override void OnSpawnServer(NetworkConnection c)
    {
        base.GiveOwnership(c);
        _st.Add(null);
    }

    private void StOnOnChange(SyncListOperation op, int index, SomeClass oldItem, SomeClass newItem, bool asserver)
    {
        Debug.Log($"{op}, {newItem == null}, {asserver}. Total count {_st.Count}");
    }
}
```

Please upload a small project reproducing this so I can take a look at that. 

https://fish-networking.gitbook.io/docs/manual/guides/creating-bug-reports#guidelines

**Comment by GasimoCodes** [comment]
I've reproduced the behaviour with this setup: 
[Repro.zip](https://github.com/user-attachments/files/17002671/Repro.zip)

Steps:
- Run server and spawn host client. The client has the right count of 1 in SyncList.
- Create another instance, join host as client.
- Server logs will say the new client has count 1. Client logs will however show count 2. 

```
using FishNet.Object.Synchronizing;
using FishNet.Object;
using UnityEngine;

public class NetTest : NetworkBehaviour
{
    class TestObj
    {
        int test;
    }

    // Fields
    private readonly SyncList<TestObj> items = new SyncList<TestObj>();

    // Methods
    public override void OnStartNetwork()
    {
        base.OnStartNetwork();
        items.OnChange += ItemList_OnChange;
    }

    public override void OnStartServer()
    {
        base.OnStartServer();
        items.Add(null);
    }


    // Actions

    private void ItemList_OnChange(SyncListOperation op, int index,
TestObj oldItem, TestObj newItem, bool asServer)
    {
        Debug.Log($"<color=green>Test items[] modified: OP: {op}, Server: {asServer}, Index: {index}, Count: {items.Count}</color>", this);

        if (op == SyncListOperation.Complete)
        {
            Debug.Log("<color=red>Operation complete. Count: " + items.Count + "</color>");
        }
    }
}
```

**Comment by Tyex** [comment]
I'm facing the same issue in my project, clients have double the count of elements in a SyncList as the server.

Everything was working fine with version 4.3.8, but today I updated to version 4.4.5R and it broke my code.

Unity version 6000.0.18f1

**Comment by FirstGearGames** [comment]
This is resolved in 4.4.6
